holy-grail-backend/backend/app/api/endpoints/library.py

```diff
@@ -15,25 +15,39 @@
 notes_router = APIRouter()
 
 
-@router.post("/", response_model=NoteSchema)
+@router.post("/", response_model=list[NoteSchema])
 @conditional_rate_limit("5/minute")
 async def create_note(
     request: Request,
-    data: NoteCreateSchema = Depends(),
-    file: UploadFile = File(None),
     authenticated: Account = Depends(Authenticator.get_verified_user),
     session: AsyncSession = Depends(get_session),
     s3_bucket: boto3.client = Depends(get_s3_client),
 ):
-    note = await Library.create(
-        session,
-        uploaded_file=file,
-        uploaded_by=authenticated.user_id,
-        data=data,
-        s3_bucket=s3_bucket,
-    )
-    new_note = await Library.get(session, note.id)
-    return new_note
+    new_notes = []
+    async with request.form() as form:
+        max_index = int(form['maxIndex'])
+        
+        for i in range(max_index + 1):
+            
+
+            form_data = NoteCreateSchema(
+                category=form[f'category {i}'],
+                subject=form[f'subject {i}'],
+                type=form[f'type {i}'],
+                document_name=form[f'document_name {i}']
+            )
+        
+            note = await Library.create(
+                session,
+                uploaded_file=form[f'file {i}'],
+                uploaded_by=authenticated.user_id,
+                data=form_data,
+                s3_bucket=s3_bucket,
+            )
+            
+            new_notes.append(await Library.get(session, note.id))
```

Comment:
From the endpoint `response_model`, you can just directly get the list of  `NoteSchema` you created from the logic above, why need to await `Library.get` again?

Also, database queries like this are very expensive, don't do that